raiden/ui/web/src/app/components/token-input/token-input.component.ts

```diff
@@ -0,0 +1,107 @@
+import { Component, Input, OnInit } from '@angular/core';
+import { FormControl, FormGroup } from '@angular/forms';
+import { MatCheckboxChange } from '@angular/material';
+import { BigNumber } from 'bignumber.js';
+import { amountFromDecimal, amountToDecimal } from '../../utils/amount.converter';
+
+
+@Component({
+    selector: 'app-token-input',
+    templateUrl: './token-input.component.html',
+    styleUrls: ['./token-input.component.css'],
+})
+export class TokenInputComponent implements OnInit {
+
+    @Input() placeholder: string;
+    @Input() errorPlaceholder: string;
+    @Input() parent: FormGroup;
+
+    private inputControl: FormControl;
+    private checkboxControl: FormControl;
+
+    constructor() {
+    }
+
+    private _decimals = 0;
+
+    public get decimals(): number {
+        return this._decimals;
+    }
+
+    public set decimals(decimals: number) {
+        this._decimals = decimals;
+        if (!this.checkboxControl) {
+            return;
+        }
+        this.updateCheckboxState();
+    }
+
+    public get tokenAmount(): BigNumber {
+        return new BigNumber(this.inputControl.value);
+    }
+
+    public get tokenAmountDecimals(): number {
+        return this.decimalInput ? this._decimals : 0;
+    }
+
+    private get decimalInput(): boolean {
+        return this.checkboxControl.value;
+    }
+
+    public resetAmount() {
+        this.inputControl.setValue('0');
+    }
+
+    ngOnInit() {
+        this.inputControl = this.parent.get('amount') as FormControl;
```

Comment:
This `parent` access adds coupling between child and parent. I'd say it'd be better, instead of inputting parent, inject the `FormControl`s directly from the parent, like: `[amountFC]="fb['amount']" [decimalsFC]="fb['decimals']"`, or alike, then `input-token` component doesn't need to know there's an `amount` and a `decimals` specifically named FormControls in the parent's FormGroup, but just that it's expecting to receive FormControls directly

Reply:
Yes, go ahead, if you're sure this one works =)